src/invert.ts

```diff
@@ -0,0 +1,49 @@
+import { purry } from './purry';
+
+type Inverted<T extends object> = T[keyof T] extends PropertyKey
+  ? Record<T[keyof T], keyof T>
+  : never;
+
+/**
+ * Returns an object whose keys are values are swapped. If the object contains duplicate values,
+ * subsequent values will overwrite previous values.
+ * @param object the object
+ * @signature
+ *    R.invert(object)
+ * @example
+ *    R.invert({ a: "d", b: "e", c: "f" }) // => { d: "a", e: "b", f: "c" }
+ * @data_first
+ * @category object
+ * @pipeable
+ */
+export function invert<T extends object>(object: T): Inverted<T>;
+
+/**
+ * Returns an object whose keys are values are swapped. If the object contains duplicate values,
+ * subsequent values will overwrite previous values.
+ * @param object the object
+ * @signature
+ *    R.invert()(object)
+ * @example
+ *    R.pipe({ a: "d", b: "e", c: "f" }, R.invert()); // => { d: "a", e: "b", f: "c" }
+ * @data_last
+ * @category object
+ * @pipeable
+ */
+export function invert<T extends object>(): (object: T) => Inverted<T>;
+
+export function invert() {
+  return purry(_invert, arguments);
+}
+
+function _invert<T extends object>(object: T): Inverted<T> {
+  const result: Record<PropertyKey, keyof T> = {};
+
+  for (const key in object) {
```

Comment:
I'm guessing that TS would have issues with `undefined` when trying to access `object[key]` because it can't tell if it exists in the input object.

I think we should use `Object.entries` here (it's not ES5, but we use it in several functions already). It would provide us a pair `[key, value]` where `value` is always well defined.

Reply:
With the other changes you mentioned, TypeScript isn't producing any errors here. I'm happy to refactor this to use `Object.entries` if you'd like, but the compiler seems happy without it.